src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -865,8 +866,9 @@ static int32_t cc_set(InteractiveConsole& console, const arguments_t& argv)
         {
             if (int_valid[0])
             {
-                const auto newClimate = int_val[0];
-                if (newClimate < 0 || newClimate >= CLIMATE_COUNT)
+                const ClimateType newClimate = ClimateType{ static_cast<uint8_t>(int_val[0]) };
```

Comment:
Technically you should validate that the `int_val[0]` is within the range of our `enum class` then if it is construct it into our new type. I.e. move this down to line 877.